src/openrct2/entity/PatrolArea.cpp

```diff
@@ -0,0 +1,177 @@
+/*****************************************************************************
+ * Copyright (c) 2014-2022 OpenRCT2 developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#include "PatrolArea.h"
+
+#include "EntityList.h"
+#include "Staff.h"
+
+static PatrolArea _consolidatedPatrolArea[EnumValue(StaffType::Count)];
+static std::variant<StaffType, EntityId> _patrolAreaToRender;
+
+static bool CompareTileCoordsXY(const TileCoordsXY& lhs, const TileCoordsXY& rhs)
+{
+    if (lhs.y == rhs.y)
+        return lhs.x < rhs.x;
+    return lhs.y < rhs.y;
+}
+
+const PatrolArea::Cell* PatrolArea::GetCell(const TileCoordsXY& pos) const
+{
+    return const_cast<PatrolArea*>(this)->GetCell(pos);
+}
+
+PatrolArea::Cell* PatrolArea::GetCell(const TileCoordsXY& pos)
+{
+    auto areaPos = TileCoordsXY(pos.x / Cell::Width, pos.y / Cell::Height);
+    if (areaPos.x < 0 || areaPos.x >= CellColumns || areaPos.y < 0 || areaPos.y >= CellRows)
+        return nullptr;
+
+    auto& area = Areas[(areaPos.y * CellColumns) + areaPos.x];
+    return &area;
+}
+
+bool PatrolArea::IsEmpty() const
+{
+    return TileCount == 0;
+}
+
+void PatrolArea::Clear()
+{
+    for (auto& area : Areas)
+    {
+        area.SortedTiles.clear();
+    }
+}
+
+bool PatrolArea::Get(const TileCoordsXY& pos) const
+{
+    auto* area = GetCell(pos);
+    if (area == nullptr)
+        return false;
+
+    auto it = std::lower_bound(area->SortedTiles.begin(), area->SortedTiles.end(), pos, CompareTileCoordsXY);
+    auto found = it != area->SortedTiles.end() && *it == pos;
```

Comment:
IMO this and the use above should be a separate function, much like this now-removed `binary_find` example from cppreference:
https://sites.radford.edu/~acm/midatl/docs/cppreference/w/cpp/algorithm/lower_bound.html

Reply:
Only for `get`, since `set` still needs to know where to insert the item. Maybe when there is a second use, we could make the trouble of a shared helper for it.